server/task_executor_pitr_cutover.go

```diff
@@ -0,0 +1,89 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/plugin/db"
+	pluginmysql "github.com/bytebase/bytebase/plugin/db/mysql"
+	restoremysql "github.com/bytebase/bytebase/plugin/restore/mysql"
+	"github.com/bytebase/bytebase/store"
+	"go.uber.org/zap"
+)
+
+// NewPITRCutoverTaskExecutor creates a PITR cutover task executor.
+func NewPITRCutoverTaskExecutor(logger *zap.Logger) TaskExecutor {
+	return &PITRCutoverTaskExecutor{
+		l: logger,
+	}
+}
+
+// PITRCutoverTaskExecutor is the PITR cutover task executor.
+type PITRCutoverTaskExecutor struct {
+	l *zap.Logger
+}
+
+// RunOnce will run the PITR cutover task executor once.
+func (exec *PITRCutoverTaskExecutor) RunOnce(ctx context.Context, server *Server, task *api.Task) (terminated bool, result *api.TaskRunResultPayload, err error) {
+	exec.l.Info("Run PITR cutover task", zap.String("task", task.Name))
+
+	// Currently api.TaskDatabasePITRCutoverPayload is empty, so we do not need to unmarshal from task.Payload.
+
+	return exec.pitrCutover(ctx, task, server)
+}
+
+func (exec *PITRCutoverTaskExecutor) pitrCutover(ctx context.Context, task *api.Task, server *Server) (terminated bool, result *api.TaskRunResultPayload, err error) {
+	driver, err := getAdminDatabaseDriver(ctx, task.Instance, "", exec.l)
+	if err != nil {
+		return true, nil, err
+	}
+	defer driver.Close(ctx)
+
+	if err := exec.doPITRCutover(ctx, task, server.store, driver); err != nil {
```

Comment:
Any reason that we need a separate function?

Reply:
Not really, removed.